### PR TITLE
Remove random beacon implementation from Narwhal

### PR DESCRIPTION
**Pull request by aschran**

### CODE REVIEW

**Review by mwtian — APPROVED** [review]

**Review by arun-koshy — APPROVED** [review]

File: narwhal/storage/src/node_store.rs [hunk 1]
```diff
@@ -120,11 +106,6 @@ impl NodeStorage {
             // This can be removed when DBMap supports removing tables.
             _sub_dag_index_map,
             committed_sub_dag_map,
-            processed_messages_map,
-            used_messages_map,
-            confirmations_map,
-            dkg_output_map,
-            randomness_round_map,
```
Comment (akichidis) [inline narwhal/storage/src/node_store.rs:127]: @aschran can we confirm that this is ok to do now? As far as I remember we can't simply delete column families (ex see that `_sub_dag_index_map` case above) when the db has been already opened for existing column families.

File: crates/sui-protocol-config/src/lib.rs [hunk 2]
```diff
@@ -1823,7 +1810,6 @@ impl ProtocolConfig {
                     }
                     // Only enable random beacon on devnet
                     if chain != Chain::Mainnet && chain != Chain::Testnet {
-                        cfg.feature_flags.narwhal_header_v2 = true;
```
[the remark this answers is not in the document]
Comment (aschran) [reply crates/sui-protocol-config/src/lib.rs:1826]: Yes, this flag only ever existed for devnet which is wiped on release so this is fine

File: narwhal/storage/src/node_store.rs (hunk 1 above)
Comment (mwtian) [reply narwhal/storage/src/node_store.rs:127]: Good call.

Comment (mwtian) [reply narwhal/storage/src/node_store.rs:127]: I think we thought about changing the rocksdb open logic across Sui and Narwhal to tolerate the issue, but decided against it for a reason.

**Pull request merged**